[Bug][Build] build-server-image failed for "/backend/python/requirements.txt" not found

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.


### What happened

build-server-image failed for "/backend/python/requirements.txt" not found
```
$ make build-server-image 
....
 => CACHED [base 3/6] WORKDIR /app                                                                                                                                                                                                                                            0.0s
 => ERROR [base 4/6] COPY backend/python/requirements.txt /app/requirements.txt                                                                                                                                                                                               0.0s
------                                                                                                                                                                                                                                                                             
 > [base 4/6] COPY backend/python/requirements.txt /app/requirements.txt:
------
ERROR: failed to solve: failed to compute cache key: "/backend/python/requirements.txt" not found: not found
Makefile:101: recipe for target 'build-server-image' failed
make[1]: *** [build-server-image] Error 1
make[1]: Leaving directory '/home/qcraft/workspace/incubator-devlake/backend'
Makefile:26: recipe for target 'build-server-image' failed
make: *** [build-server-image] Error 2
```
### What do you expect to happen

build-server-image will be success~

### How to reproduce

make build-server-image in tag: v0.16.0-beta10

### Anything else

_No response_

### Version

main

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

> This should have been fixed by #4734. @lshmouse Are you running the latest main branch?

The issue is fixed. But the build failed again with following error:
```
 => [base 7/8] RUN ln -sf /root/.local/bin/poetry /usr/local/bin                                                                                                                                                                                                              0.4s 
 => ERROR [base 8/8] RUN find ./app/python/ -name "*.sh" | xargs -I{} chmod +x {} &&     /app/python/build.sh                                                                                                                                                                 0.4s 
------                                                                                                                                                                                                                                                                             
 > [base 8/8] RUN find ./app/python/ -name "*.sh" | xargs -I{} chmod +x {} &&     /app/python/build.sh:                                                                                                                                                                            
#0 0.346 find: ‘./app/python/’: No such file or directory                                                                                                                                                                                                                          
#0 0.346 /bin/sh: 1: /app/python/build.sh: Permission denied                                                                                                                                                                                                                       
------                                                                                                                                                                                                                                                                             
ERROR: failed to solve: executor failed running [/bin/sh -c find ./app/python/ -name "*.sh" | xargs -I{} chmod +x {} &&     /app/python/build.sh]: exit code: 126
Makefile:111: recipe for target 'build-server-image' failed
make[1]: *** [build-server-image] Error 1
make[1]: Leaving directory '/home/qcraft/workspace/incubator-devlake/backend'
Makefile:26: recipe for target 'build-server-image' failed
make: *** [build-server-image] Error 2
```

I think it is a typo in backend/Dockerfile
```
-RUN find ./app/python/ -name "*.sh" | xargs -I{} chmod +x {} && \
+RUN find /app/python/ -name "*.sh" | xargs -I{} chmod +x {} && \

```

--- a/backend/Dockerfile
+++ b/backend/Dockerfile
@@ -143,7 +143,7 @@ RUN python3 -m pip install --no-cache --upgrade pip setuptools && \
 RUN curl -sSL https://install.python-poetry.org | python3 -
 RUN ln -sf /root/.local/bin/poetry /usr/local/bin
 # Build Python plugins
-RUN find ./app/python/ -name "*.sh" | xargs -I{} chmod +x {} && \
+RUN find /app/python/ -name "*.sh" | xargs -I{} chmod +x {} && \
     /app/python/build.sh
 
 

